mmv1/templates/terraform/post_create/datastream_stream.go.erb

```diff
@@ -14,4 +14,11 @@
 -%>
 if err := waitForDatastreamStreamReady(d, config, d.Timeout(schema.TimeoutCreate) - time.Minute); err != nil {
   return fmt.Errorf("Error waiting for Stream %q to be NOT_STARTED or RUNNING during creation: %q", d.Get("name").(string), err)
+}
+
+if d.Get("state") != d.Get("desired_state") {
```

Comment:
This seems to be completely unrelated to adding the other fields. Is the purpose of doing another Update call just a retry of sorts? Have you tested that this is effective?

Reply:
This is a bug fix, datastream requires an additional update call when desired_state=RUNNING